src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -325,6 +325,7 @@ internal CompilationUnitSyntax ParseCompilationUnitCore()
             }
         }
 
+        /// <summary>Are we possibly at the start of an attribute list, or at a modifier which is valid on a type, or on a keyword of a type declaration?</summary>
         private static bool IsPossibleStartOfTypeDeclaration(SyntaxKind kind)
```

Comment:
Note that the usage of this method on line 1187 (ParseModifiers) really is "is this a valid modifier for a type declaration".

It really makes one wish for reusable/composable patterns, versus being tempted to copy/paste this switch statement around.

Reply:
The only fully valid usage of this method IMO is in IsPossibleNamespaceMemberDeclaration.